src/Graph/GraphPrinter.php

```diff
@@ -133,79 +146,102 @@ public function getDependencyError() {
 	 * @return string
 	 */
 	protected function getResultText( SMWQueryResult $res, $outputmode ) {
-		global $wgVersion;
+		// Remove this once SRF requires 3.1+
+		if ( $this->hasMissingDependency() ) {
+			return $this->getDependencyError();
+		}
 
 		// iterate query result and create SRF\GraphNodes
 		while ( $row = $res->getNext() ) {
 			$this->processResultRow( $row );
 		}
 
-		// use GraphFormatter to build the graph
+		// use GraphFormater to build the graph
 		$graphFormatter = new GraphFormatter( $this->options );
 		$graphFormatter->buildGraph( $this->nodes );
 
 		// GraphViz is not working for version >= 1.33, so we need to use the Diagrams extension
 		// and formatting is a little different from the GraphViz extension
+		global $wgVersion;
 		if ( version_compare( $wgVersion, '1.33', '>=' ) &&
 			\ExtensionRegistry::getInstance()->isLoaded( 'Diagrams' ) ) {
 			$result = "<graphviz>{$graphFormatter->getGraph()}</graphviz>";
 		} else {
-			// Calls graphvizParserHook function from MediaWiki GraphViz extension
+			// Calls graphvizParserHook function from MediaWiki GraphViz or External Data extension
 			$parser = MediaWikiServices::getInstance()->getParser();
 			$result = $parser->recursiveTagParse( "<graphviz>" . $graphFormatter->getGraph() . "</graphviz>" );
 		}
 
+		// Append legend
 		$result .= $graphFormatter->getGraphLegend();
 
 		if ( $outputmode === SMW_OUTPUT_HTML ) {
 			return $result;
 		}
 
 		return MediaWikiServices::getInstance()->getParser()->recursiveTagParse( $result );
-
 	}
 
 	/**
 	 * Process a result row and create SRF\GraphNodes
 	 *
 	 * @since 3.1
 	 *
-	 * @param array $row
+	 * @param ResultArray[] $row
 	 *
 	 */
-	protected function processResultRow( array /* of SMWResultArray */ $row ) {
+	protected function processResultRow( array $row ) {
```

Comment:
Your code is still giving me a fatal error, even while I'm not using graphfields=yes.

```
[d51a7ae3f8f8465ea6880001] /index.php/Gebruiker:WikiSysop Error from line 191 of /workspace/composer-ext/amSRF/src/Graph/GraphPrinter.php: Call to a member function addField() on null

Backtrace:

#0 /workspace/composer-ext/amSRF/src/Graph/GraphPrinter.php(140): SRF\Graph\GraphPrinter->processResultRow(array)
#1 /var/platform/cores/1.35/20221110-1128/extensions/SemanticMediaWiki/src/Query/ResultPrinters/ResultPrinter.php(341): SRF\Graph\GraphPrinter->getResultText(SMW\Query\QueryResult, integer)
#2 /var/platform/cores/1.35/20221110-
```

Reply:
So far, I don't understand, how this happens within `if ( $node )`.

Reply:
I can confirm its working now. I think I was on your master branch indeed, sorry for the inconveniences. I think we've created a nice solution for now. @JeroenDeDauw can you release this branch? I'll close mine.